### PR TITLE
Chores/qiskit 1.0.2

### PR DESCRIPTION
**Pull request by gcattan**
Follow-up for #272 

Part of: #265

### CODE REVIEW

**Comment by gcattan** [comment]
The qiskit-ibm-provider dependency needs to be replaced by qiskit_ibm_runtime

**Comment by gcattan** [comment]
Need to test connection to real backend.

File: requirements.txt [hunk 1]
```diff
@@ -2,10 +2,10 @@ numpy<1.27
 cython
 scikit-learn==1.3.2
 git+https://github.com/pyRiemann/pyRiemann#egg=pyriemann
-qiskit==0.46.1
+qiskit==1.0.2
```
Comment (frankharkins) [inline requirements.txt:5]: Consider pinning to `==1.*`. One of the features of the 1.0 release is the stability guarantee that no versions of Qiskit 1.x will break compatibility. Same for the `setup.py`.

**Comment by toncho11** [comment]
I testing with the code form this PR and I get:

```
    from pyriemann_qiskit.utils.quantum_provider import get_quantum_kernel
  File "C:\Work\PythonCode\pyRiemann-qiskit\pyq_env\Lib\site-packages\pyriemann_qiskit\utils\__init__.py", line 1, in <module>
    from . import hyper_params_factory, filtering, preprocessing
  File "C:\Work\PythonCode\pyRiemann-qiskit\pyq_env\Lib\site-packages\pyriemann_qiskit\utils\hyper_params_factory.py", line 1, in <module>
    from qiskit.circuit.library import ZZFeatureMap, ZFeatureMap, PauliFeatureMap
  File "C:\Work\PythonCode\pyRiemann-qiskit\pyq_env\Lib\site-packages\qiskit\__init__.py", line 36, in <module>
    raise ImportError(
ImportError: Qiskit is installed in an invalid environment that has both Qiskit >=1.0 and an earlier version. You should create a new virtual environment, and ensure that you do not mix dependencies between Qiskit <1.0 and >=1.0. Any packages that depend on 'qiskit-terra' are not compatible with Qiskit 1.0 and will need to be updated. Qiskit unfortunately cannot enforce this requirement during environment resolution. See https://qisk.it/packaging-1-0 for more detail.
```

**Comment by toncho11** [comment]
Also I did switch `simulator` to False in order not to use a real quantum computer time.

 return provider.least_busy(
        operational=True, simulator=**False**, min_num_qubits=min_qubits
    )

**Comment by gcattan** [comment]
@toncho11 simulator is False by default. 
For your problem, I had the same. You can uninstall qiskit-terra.

**Comment by gcattan** [comment]
There is no regression in scores. Light benchmark is failing because qiskit-terra and qiskit got installed in the same container:

image

On local, to solve this error, you have to go into your environment folder and delete all `qiskit` folder, then reinsall qiskit 1.0.2.

On Ci, this happen because main is on qiskit-terra and the PR on qiskit 1.0. This will solve itself when we will merge.

**Comment by toncho11** [comment]
Sorry, I set  simulator=True in this case where I would like to use a simulated backend.

**Comment by gcattan** [comment]
I am able to get a remote instance, but I have this error message so for:

image

**Comment by gcattan** [comment]
Ok, the test dataset corresponds to 64620 circuits, and the capacity of the quantum computer is only 300 circuits per job, so I guess it is why we have so many jobs.

Changing the optimization_level for the transpiler doesn't help that much, we just need to provide less data - but this is another problem we need to address.

Here is a snapshot of the three first jobs executed in the quantum computer:

image

I think we are ready to merge. @toncho11 @frankharkins

**Comment by toncho11** [comment]
Currently I am getting:

```
from . import hyper_params_factory, filtering, preprocessing
  File "C:\Work\PythonCode\pyRiemann-qiskit\pyq_env\Lib\site-packages\pyriemann_qiskit\utils\hyper_params_factory.py", line 1, in <module>
    from qiskit.circuit.library import ZZFeatureMap, ZFeatureMap, PauliFeatureMap
ImportError: cannot import name 'ZZFeatureMap' from 'qiskit.circuit.library' (unknown location)
```

**Comment by gcattan** [comment]
Yeah. There is some namespace conflict I think between qiskit-terra and qiskit.
What worked for me was

 to go to my environment site-packages
 Delete all qiskit folders
 Reinstall qiskit 1.0.2

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by gcattan** [comment]
If you start from a clean env, it should work right away I believe.

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by toncho11** [comment]
> to go to my environment site-packages Delete all qiskit folders Reinstall qiskit 1.0.2

Yes. It worked.

**Review by toncho11 — APPROVED** [review]

**Pull request merged**